za_hansard/management/commands/za_hansard_q_and_a_scraper.py

```diff
@@ -33,6 +38,46 @@
 
 USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_3) AppleWebKit/601.4.4 (KHTML, like Gecko) Version/9.0.3 Safari/601.4.4'
 
+CODE_RE_STRING = r'R?(?P<house>[NC])(?P<answer_type>[WO])(?P<id_number>\d+)(?P<lang>[AEX])?'
+
+def tidy_for_matching(s):
+    # Sometimes the answer is accidentally included after the
+    # question. We try to exclude this in two ways. Looking for a
+    # question mark followed by a code usually means that the answer
+    # is included afterwards, so strip that out:
+    s = re.sub(r'^(?ims)(.*\?)\s*' + CODE_RE_STRING + r'.*', r'\1', s)
+    # Also, sometimes that an answer is accidentally included can be
+    # spotted by a line of text that has no lower case "QUESTION
+    # REPLY", etc. so strip out anything after one of them:
+    lines = s.splitlines()
+    break_found = False
+    for i, l in enumerate(lines):
+        l = l.strip()
+        if re.search(r'^[^a-z]+$', l):
+            break_found = True
+            break
+    if break_found:
+        result = '\n'.join(lines[:i])
+    else:
+        result = s
+    return re.sub(r'(?ims)\s+', ' ', result).strip()
+
+def text_similarity(a, b):
+    return SequenceMatcher(None, a, b).ratio()
+
+def find_matching_question(questions_qs, target):
+    q_list = [
+        (text_similarity(
+            tidy_for_matching(question.question),
+            tidy_for_matching(target)),
+         question)
+        for question in questions_qs
+    ]
+    if not q_list:
+        raise Question.DoesNotExist()
+    ratio, question = min(q_list)
```

Comment:
It is not clear to me why min works here given the second value in the tuple is a question object.